### DISCUSSION TITLE
Can arm64 ever run in desktop mode?

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#10733] [General]
Avalonia has two modes, Desktop or SingleView.

Are there cases where compiling for arm64 (or linux-arm64) would run in desktop mode? I note that there's a Windows arm64 version; any other case? Ah ya, Apple M1/M2 chips are also arm64.

But on Linux, where is `linux-arm64` architecture used?

**Comment by kekekeks** [comment]
The "mode" doesn't depend on CPU architecture, it depends on the OS and the way the OS is configured (in case of fbdev/drm on Linux)

**Reply by mysteryx93** [reply]
I'm curious what Linux OS runs on arm64 architecture

**Comment by ahopper** [comment]
Raspberry pi 64bit os

**Reply by mysteryx93** [reply]
Raspberry pi is for compact device; wouldn't that run in SingleView mode?

**Reply by ahopper** [reply]
It could but most of my users run on rpi plugged into big monitors in desktop mode.

**Comment by mysteryx93** [comment]
Also, when I get to publish an Android app, do I need to publish android-x64 or only android-arm64 ?

Any point in publishing an android-x64 and iOS-x64 app?

(Bass audio released no win-arm64 so far because he saw no point in it yet)

**Reply by mysteryx93** [reply]
This makes it pretty clear that native binaries need to be published for various architectures.
https://devblogs.microsoft.com/xamarin/64-bit-requirements-xamarin-android-apps/

Which ones do Avalonia support for Android and iOS?

**Reply by mysteryx93** [reply]
OK it has Xamarin's limitations
https://learn.microsoft.com/en-us/xamarin/android/app-fundamentals/cpu-architectures?tabs=windows

Looks like I have to publish for all 4 Android architectures and 2 iOS architectures then?

**Reply by maxkatz6** [reply]
> Which ones do Avalonia support for Android and iOS?

There is no difference for Avalonia.

> Looks like I have to publish for all 4 Android architectures and 2 iOS architectures then?

There should be official apple store and google play recommendations somewhere...
